src/web/mjs/connectors/MangaHost.mjs

```diff
@@ -28,73 +39,50 @@ export default class MangaHost extends Connector {
+                title: element.text.trim()
+            };
+        });
+        if (index < mangaPageLinks.length - 1) {
+            return this._getMangaListFromPages(mangaPageLinks, index + 1)
+                .then(mangas => mangas.concat(mangaList));
+        } else {
+            return mangaList;
+        }
     }
 
-    _getMangaList(callback) {
-        this.fetchDOM(this.url + '/mangas', 'div.paginador div.wp-pagenavi a.last')
-            .then(data => {
-                let pageCount = parseInt(data[0].href.match(/(\d+)$/)[1]);
-                let pageLinks = [... new Array(pageCount).keys()].map(page => this.url + '/mangas/page/' + (page + 1));
-                return this._getMangaListFromPages(pageLinks);
-            })
-            .then(data => {
-                callback(null, data);
-            })
-            .catch(error => {
-                console.error(error, this);
-                callback(error, undefined);
-            });
+    async _getMangas() {
+        const data = await this.fetchDOM(this.url + '/mangas', 'div.paginador div.wp-pagenavi a.last');
+        let pageCount = parseInt(data[0].href.match(/(\d+)$/)[1]);
+        let pageLinks = [... new Array(pageCount).keys()].map(page => this.url + '/mangas/page/' + (page + 1));
+        return this._getMangaListFromPages(pageLinks);
     }
 
-    _getChapterList(manga, callback) {
+    async _getChapters(manga) {
         const url = new URL(manga.id, this.url);
         url.searchParams.set('t', Date.now());
-        this.fetchDOM(url.href, 'div.chapters div.card.pop div.pop-title span')
-            .then(data => {
-                let chapterList = data.map(element => {
-                    return {
-                        id: [manga.id, element.textContent].join('/'),
-                        title: element.textContent,
-                        language: 'pt'
-                    };
-                });
-                callback(null, chapterList);
-            })
-            .catch(error => {
-                console.error(error, manga);
-                callback(error, undefined);
-            });
+        const data = await this.fetchDOM(url.href, 'div.chapters div.card.pop div.pop-title span');
+        return data.map(element => {
+            return {
+                id: [manga.id, element.textContent].join('/'),
+                title: element.textContent,
+                language: 'pt'
+            };
+        });
     }
 
-    _getPageList(manga, chapter, callback) {
-        let request = new Request(this.url + chapter.id, this.requestOptions);
-        this.fetchRegex(request, /<img\s+id='img_\d+'\s+src='(.*?)'/g)
-            .then(data => {
-                callback(null, data);
-            })
-            .catch(error => {
-                console.error(error, chapter);
-                callback(error, undefined);
-            });
+    async _getPages(chapter) {
+        console.log(chapter.id);
```

Comment:
Production code should not dump debug information

Reply:
original code :\ didn't rework much. I guess i'll have to clean someone else code :'(